features/consent/update-groups-consent-for-account.ts

```diff
@@ -0,0 +1,43 @@
+import logger from "@utils/logger";
+import { ConverseXmtpClientType } from "@/utils/xmtpRN/client.types";
+import { getXmtpClient } from "@/utils/xmtpRN/sync";
+
+type UpdateGroupsConsentForAccountArgs = {
+  account: string;
+  groupIds: string[];
+  consent: "allow" | "deny";
+};
+
+export const updateGroupsConsentForAccount = async (
+  args: UpdateGroupsConsentForAccountArgs
+) => {
+  const { account, groupIds, consent } = args;
+
+  const client = (await getXmtpClient(account)) as ConverseXmtpClientType;
+
+  if (!client) {
+    throw new Error("Client not found");
+  }
+
+  logger.debug(
+    `[XMTPRN Contacts] Consenting to groups on protocol: ${groupIds.join(", ")}`
+  );
+  const start = new Date().getTime();
+
+  const state = consent === "allow" ? "allowed" : "denied";
```

Comment:
do this in the for users one but we also probably want to throw if invalid consent, but that'd be a typescript issue anyway so you're choice whether you want to extract this to a function or not